### PR TITLE
Refactor Authors, Sources and Tags classes

### PR DESCRIPTION
**Pull request by H4mes**
Added classes for author, source and tag.

Added tag pill UI elements for authors, sources and tags.

### CODE REVIEW

**Review by Ko-Khan — COMMENTED** [review]
LGTM, but maybe check the tests to ensure they are still working, as you changed a lot of old code

**Review by bennyLCK — COMMENTED** [review]
Some possible flaws to rectify are present.

File: src/main/java/seedu/address/logic/parser/EditArticleCommandParser.java [hunk 1]
```diff
@@ -73,30 +76,34 @@ public EditArticleCommand parse(String args) throws ParseException {
...
-        for (String author : authors) {
-            if (author.isEmpty()) {
-                return Optional.empty();
-            }
+        if (authors.isEmpty()) {
+            return Optional.empty();
         }
-        List<String> authorSet = authors;
-        authorSet = authorSet.size() == 1 && authorSet.contains("") ? Collections.emptyList() : authorSet;
+        Collection<String> authorSet = authors.size() == 1 && authors.contains("") ? Collections.emptySet() : authors;
```
[the remark this answers is not in the document]
Comment (H4mes) [reply src/main/java/seedu/address/logic/parser/EditArticleCommandParser.java:85]: Thanks for catching that. To clarify, the intent is to eventually have authors link to a contact in the addressbook. In cases where the user is the sole author of an article, should they be able to add an article with no author or instead required to add themselves as a contact and reference their own information?

[the remark this answers is not in the document]
Comment (bennyLCK) [reply src/main/java/seedu/address/logic/parser/EditArticleCommandParser.java:85]: Thanks for clarifying the intention of  having authors linked to Person objects in addressbook.

**Review by bennyLCK — APPROVED** [review]
LGTM

**Pull request merged**